From 95a2b50841648c06bacca7b3d37a3b99d4f71d1e Mon Sep 17 00:00:00 2001
From: Theo Buehler <botovq@users.noreply.github.com>
Date: Tue, 22 Nov 2022 16:33:32 +0100
Subject: [PATCH] LibreSSL 3.7 adds support for Ed25519 (#7803)

(tweaked slightly; py-cryptography dropped OpenSSL 1.1 support on main
but not the 38.0.x branch)

Index: src/_cffi_src/openssl/evp.py
--- src/_cffi_src/openssl/evp.py.orig
+++ src/_cffi_src/openssl/evp.py
@@ -215,9 +215,10 @@ static const long Cryptography_HAS_ONESHOT_EVP_DIGEST_
 #endif
 
 #if CRYPTOGRAPHY_OPENSSL_LESS_THAN_111
-static const long Cryptography_HAS_RAW_KEY = 0;
 static const long Cryptography_HAS_EVP_DIGESTFINAL_XOF = 0;
 int (*EVP_DigestFinalXOF)(EVP_MD_CTX *, unsigned char *, size_t) = NULL;
+#if CRYPTOGRAPHY_LIBRESSL_LESS_THAN_370 || !CRYPTOGRAPHY_IS_LIBRESSL
+static const long Cryptography_HAS_RAW_KEY = 0;
 EVP_PKEY *(*EVP_PKEY_new_raw_private_key)(int, ENGINE *, const unsigned char *,
                                        size_t) = NULL;
 EVP_PKEY *(*EVP_PKEY_new_raw_public_key)(int, ENGINE *, const unsigned char *,
@@ -227,6 +228,9 @@ int (*EVP_PKEY_get_raw_private_key)(const EVP_PKEY *, 
 int (*EVP_PKEY_get_raw_public_key)(const EVP_PKEY *, unsigned char *,
                                    size_t *) = NULL;
 #else
+static const long Cryptography_HAS_RAW_KEY = 1;
+#endif
+#else
 static const long Cryptography_HAS_RAW_KEY = 1;
 static const long Cryptography_HAS_EVP_DIGESTFINAL_XOF = 1;
 #endif
